### PR TITLE
Bookmark endpoints

### PR DESCRIPTION
**Pull request by mkotik**
## Description

contributors:

-Bradford Meziere
-Connor Rayburn
-Marat Kotik
-Fernando Velasco

https://www.loom.com/share/7beb6bcea2c64ad29f94c9731e2bf431

Fixes # (issue)

-added addBookmark model function
-add route to post a new bookmark
-fixed a prettier bug (deleted "--eslint" in the configuration file)
-changed "userId" to "id" in the bookmarks table migration file and seed file

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ x] This change requires a documentation update

## Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ x] I have removed unnecessary comments/console logs from my code
- [ x] I have made corresponding changes to the documentation if necessary (optional)
- [ x] My changes generate no new warnings
- [ x] I have checked my code and corrected any misspellings
- [ x] No duplicate code left within changed files
- [ x] Size of pull request kept to a minimum
- [ x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by JackOrth — APPROVED** [review]
Thanks for doing this! These are great clean-ups, and very necessary changes.

**Review by RimshaSaleem — APPROVED** [review]

**Review by bevanger — APPROVED** [review]
Great job adding those bookmark endpoints, the code looks great. And good job catching that bug and fixing it.

File: api/bookmarks/bookmarksModel.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const getbookmarks = () => {
+  return db('bookmarks');
+};
+
+const getByUserId = async (id) => {
+  return await db('bookmarks').where('id', id);
+};
+
+const addbookmark = async (bookmark) => {
+  await db('bookmarks').insert(bookmark);
```
Comment (collerhatguy) [inline api/bookmarks/bookmarksModel.js:12]: I'm pretty sure you can just return this line and that would return the newly created bookmark. heres some code where i did the sam thing: https://github.com/Build-Week-Potluck-Planner-4/back-end/blob/main/api/potlucks/modal.js

**Review by collerhatguy — APPROVED** [review]
Can't wait to connect these to the backend! love the detail and work that went into this. Tests aren't fun to write but you did it anyway.

**Review by juliane1999 — APPROVED** [review]

**Review by caralocke — APPROVED** [review]
Looks great to me! Can't wait to get to see them in action!

**Review by engegreg — APPROVED** [review]
looks extremely useful, great code.

**Review by nora-exe — APPROVED** [review]
Marat & friends - Your additions look clean, DRY, and modular/reusable. Some minor merge conflicts. Clear simple Loom walkthrough. LGTM 👍

**Review by bverfurth — APPROVED** [review]
Excellent job cleaning this up!

**Review by rkshockey — APPROVED** [review]
Everything looks good to me. The only thing that seems to be missing is a way to remove bookmarks.

**Review by fervelgo — APPROVED** [review]
Ready ready ready handle those bookmarks!

**Comment by tdubs42** [comment]
[Roles table in PostrgeSQL db and FE auth](https://trello.com/c/hzgYJgPB/83-roles-table-in-postrgesql-db-and-fe-auth)

**Pull request merged**